lldb/include/lldb/Core/ValueObject.h

```diff
@@ -719,6 +776,10 @@ class ValueObject {
     ClearUserVisibleData(eClearUserVisibleDataItemsSummary);
   }
 
+  void SetDerefValobj(ValueObject *deref) { m_deref_valobj = deref; }
```

Comment:
Synthetic child providers also have a way to provide the deref ValueObject dynamically.  How do these two ways of doing that job work together?

Reply:
I use the synthetic child providers way of getting the deref ValueObject  to get the value I use for my call to SetDerefValobj (when I'm converting a ValueObject containing a SmartPointer into a ValueObject containing the pointer the SmartPointer was referencing).